lib/Supervisor.js

```diff
@@ -29,22 +30,29 @@ function run(
     var workers = [];
 
     function printResult(result, exitCode /*: number | void */ = undefined) {
-      if (format === 'console') {
-        // todos are objects, and will be shown in the SUMMARY only.
-        // passed tests are nulls, and should not be printed.
-        // failed tests are strings.
-        if (typeof result === 'string') {
-          const message = makeWindowsSafe(result);
-          if (exitCode === 1) {
-            console.error(message);
-          } else {
-            console.log(message);
+      switch (report) {
+        case 'console':
+          // todos are objects, and will be shown in the SUMMARY only.
+          // passed tests are nulls, and should not be printed.
+          // failed tests are strings.
```

Comment:
Can we use flow to enforce this? No worries if not.

Reply:
Supervisor.js is one of the parts of node-test-runner I haven’t grasped yet. I’m not familiar enough with this piece of code to make further changes to it yet – all I did here was to replace `if-else` with a `switch` over the possible `Report` variants.

Reply:
That makes perfect sense